modules/cms-kit/src/Volo.CmsKit.Admin.Application/Volo/CmsKit/Admin/Blogs/BlogPostAdminAppService.cs

```diff
@@ -110,4 +113,30 @@ public virtual async Task DeleteAsync(Guid id)
     {
         await BlogPostRepository.DeleteAsync(id);
     }
-}
+
+    [Authorize(CmsKitAdminPermissions.BlogPosts.Publish)]
+    public virtual async Task PublishAsync(Guid id)
+    {
+        var blogPost = await BlogPostRepository.GetAsync(id);
+        blogPost.SetPublished();
+    }
+
+    [Authorize(CmsKitAdminPermissions.BlogPosts.Update)]
+    public virtual async Task DraftAsync(Guid id)
+    {
+        var blogPost = await BlogPostRepository.GetAsync(id);
+        blogPost.SetDraft();
+    }
+
+    [Authorize(CmsKitAdminPermissions.BlogPosts.Create)]
+    [Authorize(CmsKitAdminPermissions.BlogPosts.Publish)]
+    public virtual async Task<BlogPostDto> CreateAndPublishAsync(CreateBlogPostDto input)
+    {
+        var blogPost = await CreateAsync(input);
+        await CurrentUnitOfWork.SaveChangesAsync();
+        
+        await PublishAsync(blogPost.Id);
+        blogPost.Status = BlogPostStatus.Published;
```

Comment:
Duplicated! 
`blogPost.Status = BlogPostStatus.Published;` it has already been seted in `await PublishAsync(blogPost.Id);`

Reply:
No is it not. It is dto which values filled before we change entity's status value.

Reply:
ok,I get it

Reply:
Thank you for taking the time to contribute @rqx110